src/RA_StringUtils.h

```diff
@@ -329,41 +333,59 @@ class StringBuilder
         while (*pFormat)
         {
             auto* pScan = pFormat;
+            if (pScan == nullptr)
+            {
+                ::MessageBox(::GetActiveWindow(), _T("Invalid format string!"), _T("Error!"), MB_OK | MB_ICONERROR);
+                break;
+            }
             while (*pScan && *pScan != '%')
                 ++pScan;
 
             if (*pScan == '%')
             {
                 ++pScan;
                 AppendSubString(pFormat, pScan - pFormat);
-
                 if (*pScan == '%')
+                {
                     ++pScan;
+                }
                 else if (*pScan != '\0')
+                {
                     assert(!"not enough parameters provided");
+                }
             }
             else if (pScan > pFormat)
             {
                 AppendSubString(pFormat, pScan - pFormat);
             }
-
             pFormat = pScan;
         }
     }
 
     template<typename CharT, typename = std::enable_if_t<is_char_v<CharT>>, typename T, typename... Ts>
-    void AppendPrintf(const CharT* const pFormat, const T& value, Ts&&... args)
+    void AppendPrintf(const CharT* const restrict pFormat, const T& restrict value, Ts&&... args)
     {
         auto* pScan = pFormat;
+        if (pScan == nullptr)
+        {
+            ::MessageBox(::GetActiveWindow(), _T("Invalid format string!"), _T("Error!"), MB_OK | MB_ICONERROR);
```

Comment:
`MessageBox` should not be called directly. Use `MessageBoxViewModel` instead. However, in this particular case, I don't think it should be called at all. The user can't do anything with this, so why tell them? Recommend guarding with an `assert` or `Expects`.

Reply:
I used the ViewModel in other places where MessageBox was used